### DISCUSSION TITLE
TeslaMate - Still active? or Dead?

### DISCUSSION THREAD

**Discussion by defconxt** [teslamate-org/teslamate#3396] [Polls]
Looks like the last commit was July, 138 open issues? I've been using TeslaFi myself, but since then built a small homeserver a few weeks ago and had this bookmarked to deploy at some point...is it worth it / in a working state? It would be cool to save the $5/mo, but I don't want to deploy something that is no longer maintained and self host it.

**Comment by djpalmis** [comment]
Project Owner doesn't even bother to reply to https://github.com/adriankumpf/teslamate/discussions/3384 so.. I guess he lost interest or whatsoever.

I hope there will be a new project starting from this.

**Comment by parkr** [comment]
While active development seems to be on pause, Teslamate continues to be a fantastic piece of software that works well for me every day. Simply because regular updates are no longer occurring doesn't automatically make it worthless to deploy IMO. Its current restrictions will likely stay for now, but I have not had significant issues that require updates myself, so I am happy with the latest release.

**Reply by cwanja** [reply]
> Teslamate continues to be a fantastic piece of software that works well for me every day
>
The challenge with that statement is TeslaMate seems to break with a recent Tesla update that changes the Tesla API. See #3371, #3370, #3389 and #3386. There is a "fix", but that fix is not working for every underlying OS that can run TeslaMate, it is unofficial and not as easy to install as a vanilla TeslaMate. 

Here is to hoping that @adriankumpf will update TeslaMate or at least let us know that the current version is the last he will support.

**Reply by imthefrizzlefry** [reply]
I agree this is a wonderful piece of software, but the current official release is unusable by 10% of tesla owners, and that number is growing by hundreds of installs every day.   There are now several forks popping up to compensate for a lack of basic maintenance to keep the software usable.

**Reply by cwanja** [reply]
It is / was a known side effect of TeslaMate from the beginning. This was never commercial software and Adrian never said he supports it full-time. It has been a community effort and continues to be. However, from my perspective, Adrian has ceased development from his repo. 

Truthfully, I do not think the massive hack earlier this year helped. I put little to no responsibility on Adrian for that, but I assume he is a software developer professionally and does not want that hanging over him.

**Reply by djpalmis** [reply]
I don't understand what is the massive hack you are taking, can you explain me?
Anyway: first of all I wish the best to the project owner, because We don't know if he has some personal problem (health, job, relationship and so on). 
So I wish him that he is so happy that he don't have time to spend on this project! 

But after that we need to move organically on just one (sadly) unofficial fork Instead of hundreds of forks!

**Reply by cwanja** [reply]
> I don't understand what is the massive hack you are taking, can you explain me?
> 

https://medium.com/@david_colombo/how-i-got-access-to-25-teslas-around-the-world-by-accident-and-curiosity-8b9ef040a028

Basically exposed Grafana instances and then the ability to get tokens and run commands.

**Reply by djpalmis** [reply]
:-o I didn't know anything. Ehm.. are we still in danger?

**Reply by cwanja** [reply]
> :-o I didn't know anything. Ehm.. are we still in danger?

Decision tree is:
1. Is your instance exposed to the internet? Can you get to it outside your home network or where ever it is hosted? If no, stop. 
2. If yes, have you updated to the latest release and encrypted your Tesla tokens? If yes, stop. 
3. If no, update and encrypt your tokens. Cycle your Tesla credentials for good measure.

**Reply by djpalmis** [reply]
Thank you so much for your precious support!

**Reply by aiden-cat** [reply]
IMO that "hack" is not an issue at all.  I don't get how you could be savvy enough to host it on google cloud BUT then not set a secure password.  Ridiculous

**Comment by cwanja** [comment]
#3384 Adrian has spoken. He will not further development, but is open to forks.
